### ISSUE TITLE
about detection score

### ISSUE THREAD

**Issue by William-oliver** [alinlab/CSI#20]
hi, sir.
the test score function make me confuse.
1.
Eq.(7) utilizing shifting transformations S upon the detection score Eq.(6) that is applicable to any contrastive representation. Given a training data B={x_1, x_2} and set S = {s_0=I, s_1}. then we  identify whether x belong to the in-distribution or not.
In Eq.(7),  it is not only find the maximum cosine similarity between sim(x_1, x) and sim(x_2, x), but also find the maximum cosine similarity from  the transformations of x    and    the transformations of B  (e.g., sim(s_1(x_1), s_1(x)) and sim(s_1(x_2), s_1(x))  ), that make me confuse: why we need the maximum similarity between transformations of x    and    the transformations of B ? What is the meaning of this？

2.
Eq.(8) utilizing the auxiliary classifier p(yS |x) upon fθ 。what is the output of Eq.(8) ? and I don't understand what does this formula represent ? 
Eq.(9) = Eq.(7) + Eq.(8). what does this formula represent ? similarity between test data  x  and nearest training data(x_1 or x_2)?   Do we need to set a threshold for the actual application?

3.
You said that we hypothesize that the SimCLR objective implicitly increases the norm, as it is hard to decrease the Euclidian distance of the features u and v. so, it is  just  hypothesize ???   
the paper said :"We suspect that increasing the norm may be an easier way to maximize cosine similarity between two vectors". but, in fact the cosine similarity should be decrease when we increasing the norm?

**Comment by sangwoomo** [comment]
Hi, @William-oliver,

1. If x is in-distribution with respect to the original distribution, s(x) would also be in-distribution with respect to the transformed distribution. Here, sim(s_1(x_1), s_1(x)) measures the OOD-ness of s_1(x_1) with respect to the transformed distribution. Intuitively, we ensemble the detection score over all shifting transformations.

2. Most prior work uses the softmax probability as a detection score. Eq. (8) is just a logit values (before softmax) of the auxiliary classifier, which resembles the prior work. Eq. (9) simply aims to combine two scores. One need to set a threshold, which can be computed from the training data; it trade-offs the precision-recall (a.k.a. ROC curve).

3. SimCLR objective aims to decrease the cosine similarity of z1 and z2. Here, one can decrease the absolute distance of z1 and z2, or increase the scale of both to decrease the relative scale of the absolute distance. We hypothesize that the later one is an easier way to decrease the cosine similarity.

**Comment by William-oliver** [comment]
thanks for your quick reply.
